src/test/java/pl/java/spring/web/project/clientappproject/login/controller/UserValidatorMockTest.java

```diff
@@ -0,0 +1,52 @@
+package pl.java.spring.web.project.clientappproject.login.controller;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.Errors;
+import org.springframework.validation.FieldError;
+
+import java.util.Arrays;
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+@SpringBootTest
+public class UserValidatorMockTest {
+
+    @InjectMocks
+    private UserValidator userValidator;
+
+    @Mock
+    private UserService userServiceMock;
+
+    @Test
+    public void userAlreadyExists() {
+        when(userServiceMock.findByUsername("faplllosss")).thenReturn(new User());
+
+        User invalidUser = new User();
+        invalidUser.setId(1L);
+        invalidUser.setUsername("faplllosss");
+        invalidUser.setPassword("faplllosss123");
+        invalidUser.setPasswordConfirm("faplllosss123");
+
+        Errors errors = new BeanPropertyBindingResult(invalidUser, "invalidUser");
+        userValidator.validate(invalidUser, errors);
+
+        FieldError usernameFieldError = errors.getFieldError("username");
+        assertEquals("username", usernameFieldError.getField());
```

Comment:
Instead of using assertEquals try to use assertThat( "actual", is("expected")). It is just more readable. It is really easy to forgot what is expected and what is actual in assertEquals assertions